src/sentry/rules/processor.py

```diff
@@ -276,7 +278,7 @@ def activate_downstream_actions(
             action_inst = action_cls(self.project, data=action, rule=rule)
 
             results = safe_execute(
-                action_inst.after,
+                action_inst.after,  # type: ignore # passes changes depending on the action type, which each have their own class that inherits from EventAction, which itself inherits from RuleBase
```

Comment:
type ignores should only be used for bugs in mypy and they should always include a code.  while enabling a file for type checking it should be improved where possible so it passes type checking -- not just ignoring the errors

Reply:
gotcha, see my comment here: https://github.com/getsentry/sentry/pull/58281#discussion_r1362685787

what would you recommend doing to fix the mypy issues? is there a way to cast an instance of a parent class to a child class?

Reply:
you could use `Self` type probably?  but it may indicate that this should be refactored to be typesafe

Reply:
hm i actually have an idea, i'm going to push a fix and re-request your review! thank you!

Reply:
what does `reveal_type` give on each of these?

Reply:
ugh tests are failing now with the change i introduced, i'll do more digging